cpp/src/metrics/silhouette_score_batched_double.cu

```diff
@@ -0,0 +1,44 @@
+
+/*
+ * Copyright (c) 2021-2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuml/metrics/metrics.hpp>
+#include <raft/distance/distance_types.hpp>
```

Comment:
It would be worth adding this to the includes and seeing if it improves the build time at all (without having to make changes to RAFT):

```
#if defined RAFT_DISTANCE_COMPILED
#include <raft/distance/specializations.cuh>
#endif
```

Reply:
Even the RAFT batched API invokes the RAFT pairwise distance API, which is by far the largest repeat offener of large build times in both raft and cuML: https://github.com/rapidsai/raft/blob/bb888f48e6bbfbb39c0749f082bae544d2773339/cpp/include/raft/stats/detail/batched/silhouette_score.cuh#L142